caddyhttp/proxy/upstream.go

```diff
@@ -415,23 +442,47 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream, hasSrv bool) error {
 		}
 		u.HealthCheck.ContentString = c.Val()
 	case "header_upstream":
-		var header, value string
-		if !c.Args(&header, &value) {
-			// When removing a header, the value can be optional.
-			if !strings.HasPrefix(header, "-") {
+		var header, value, replaced string
+		if c.Args(&header, &value, &replaced) {
+			// Don't allow - or + in replacements
+			if strings.HasPrefix(header, "-") || strings.HasPrefix(header, "+") {
+				return c.ArgErr()
+			}
+			r, err := regexp.Compile(value)
+			if err != nil {
 				return c.ArgErr()
```

Comment:
Should consider returning the compilation error to help debug regexes. (Below too)

Reply:
Good idea, done.